core/src/data_sources/node.rs

```diff
@@ -1,19 +1,91 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+use crate::project::Project;
+
+#[derive(Debug, Clone, Serialize, PartialEq, Deserialize, Copy)]
 pub enum NodeType {
     Document,
     Table,
     Folder,
 }
 
```

Comment:
I think we'll have a simpler and less verbose interface with:
```
pub struct Node {
...
}

pub trait ToNode {
  fn node(&self) -> Node
}
```

The fact that we need the "SimpleNode" name is a good sign that we're doing something not quite right here IMHO

Reply:
The interface can return `Node` really here as it's a very small object. It will make our life much easier. But if there is strong feeling we can also do `&Node` but I think it will be slightly less convenient.